### PR TITLE
Remove unused scss variables

### LINKED ISSUE #21551: Remove unused variables
These variables are defined, but not utilized. Remove all or advise which ones to keep.
```
$inverse-bg
$inverse-color
$font-size-xs
$abbr-border-color
$btn-toolbar-margin
$custom-control-indicator-margin-y
$zindex-navbar
$navbar-border-radius
$navbar-brand-padding-y
$nav-item-margin
$nav-item-inline-spacer
$nav-tabs-justified-link-border-color
$nav-tabs-justified-active-link-border-color
$progress-box-shadow
$pre-bg
$pre-border-color
```

### PR DESCRIPTION
**Pull request by pvdlg**
Fixes #21551 

Remove unused scss variables mentioned in #21551 and a few more.

### CODE REVIEW

**Comment by Quy** [comment]
On this page too. Thanks.
https://v4-alpha.getbootstrap.com/layout/overview/

**Comment by vanduynslagerp** [comment]
Done. Good catch @Quy !

File: scss/_variables.scss [hunk 1]
```diff
@@ -851,7 +832,6 @@ $progress-height:               1rem !default;
 $progress-font-size:            .75rem !default;
 $progress-bg:                   $gray-lighter !default;
 $progress-border-radius:        $border-radius !default;
-$progress-box-shadow:           inset 0 .1rem .1rem rgba($black,.1) !default;
```
Comment (mdo) [inline scss/_variables.scss]: This one can remain; I'll need to add a conditional shadow to the `.progress` to make use of it.

Comment (vanduynslagerp) [inline scss/_variables.scss]: Done

**Comment by vanduynslagerp** [comment]
Re-added $progress-box-shadow

**Pull request merged**